source/agent/conference/conference.js

```diff
@@ -1893,14 +1893,22 @@ var Conference = function (rpcClient, selfRpcId) {
     if ((room_id === roomId) && roomController) {
       if (typeof target.view === 'string') {
         const view = target.view;
+        const input = streams[activeInputStream] ?
+            activeInputStream : trackOwners[activeInputStream];
         room_config.views.forEach((viewSettings) => {
           if (viewSettings.label === view && viewSettings.video.keepActiveInputPrimary) {
-            roomController.setPrimary(activeInputStream, view);
+            if(streams[input].info.type === "webrtc") {
+                const videoTrackId = streams[input].media.tracks.filter(t => t.type === 'video').map(t => t.id).toString();
```

Comment:
map(...).toString() converts a Array to string, if the Array contains more than 1 items(current simulcast streams have multiple video tracks), this has potential problems. You will pass a string like 'id1,id2' to setPrimary.
Use Array.find() or [0] to get exactly one element in the Array instead of calling Array.toString.